framework/PostProcessors/DataMining.py

```diff
@@ -347,40 +347,118 @@ def _localReadMoreXML(self, xmlNode):
       self.raiseAnError(RuntimeError, 'No available Output to collect (Run probably is not finished yet)')
-    dataMineDict = finishedJob.getEvaluation()[1]
+
+    inputObject, dataMineDict = finishedJob.getEvaluation()
+
+    ## This should not have to be a list
+    ## TODO: figure out if there is a case where it can be in this processor
+    inputObject = inputObject[0]
+
+    ## Store everything we cannot wrangle from the input data object and the
+    ## result of the dataMineDict
+    missingKeys = []
+
+    ############################################################################
+    ## If the input data object is different from the output data object, we
+    ## need to explicitly copy everything over that the user requests from the
+    ## input object
+    if inputObject != outputObject:
+
+      ## First copy any data you need from the input object
+      ## before adding new data
+      if outputObject.type == 'PointSet':
+        for key in outputObject.getParaKeys('Input')+outputObject.getParaKeys('Output'):
+          column = None
+          if key in inputObject.getParaKeys('Input'):
+            column = copy.copy(inputObject.getParam('Input', key))
+          elif key in inputObject.getParaKeys('Output'):
+            column = copy.copy(inputObject.getParam('Output', key))
+
+          if column is None:
+            missingKeys.append(key)
+          else:
+            for value in column:
+              if key in outputObject.getParaKeys('Input'):
+                outputObject.updateInputValue(key, value)
+              else:
+                outputObject.updateOutputValue(key, value)
+      elif outputObject.type == 'HistorySet':
+        ## This should really be fixed. Copying from a HistorySet should not
+        ## have a different use of the same API as copying a PointSet, the
+        ## way this data is stored internally seems to be bleeding over into
+        ## other parts of the code making everything else less maintainable
+        ## -- DPM 7/Aug/2017
+        for key in outputObject.getParaKeys('Input')+outputObject.getParaKeys('Output'):
+          column = None
+          if key in inputObject.getParaKeys('Input'):
+            column = {}
+            for historyIdx in inputObject.getParametersValues('inputs').keys():
+              column[historyIdx] = copy.copy(inputObject.getParam('Input', historyIdx)[key])
+            # column = copy.copy(inputObject.getParam('Input', key))
+          elif key in inputObject.getParaKeys('Output'):
+            column = {}
+            for historyIdx in inputObject.getParametersValues('outputs').keys():
+              column[historyIdx] = copy.copy(inputObject.getParam('Output', historyIdx)[key])
+            # column = copy.copy(inputObject.getParam('Output', key))
+
+          if column is None:
+            missingKeys.append(key)
+          else:
+            for historyIdx,value in column.items():
+              if key in outputObject.getParaKeys('Input'):
```

Comment:
Here is where we determine where the extracted data from the input object goes into the output object.